test/jit/test_list_dict.py

```diff
@@ -408,6 +408,34 @@ def test_over_slice():
             return a[3:10] == [3, 4]
         self.checkScript(test_backward_slice, ())
 
+    def test_slice_index(self):
+        def test_func(fn, inputs):
+            self.assertEqual(fn(*inputs), torch.jit.script(fn)(*inputs))
+
+        a = torch.tensor(
+            [
+                [[1, 11], [2, 22]],
+                [[3, 33], [4, 44]],
+                [[5, 55], [6, 66]],
+            ]
+        )
+
+        def test_index_slice1(x):
+            x = x[:, :, [0, 1]]
+            return x
+
+        def test_index_slice2(x):
+            x = x[[2, 1, 0], :, :]
+            return x
+
+        def test_index_slice3(x):
+            x = x[[0, 1], :, [1]]
+            return x
+
+        test_func(test_index_slice1, (a,))
```

Comment:
You can use self.checkScript for same functionality.

Reply:
Ah, I didn't use that because I wasn't sure if it was comparing the outputs of the scripted vs non-scripted modules. I double checked on the self.checkScript code and I see that it actually does. I'll change it to use that instead :)